components/server/src/workspace/workspace-starter.ts

```diff
@@ -345,33 +346,48 @@ export class WorkspaceStarter {
                 auth.setSelective(selectiveAuth);
             }
             if (WorkspaceImageSourceDocker.is(imgsrc)) {
-                // We only build Docker images from Git initializer. To ensure that createGitInitializer relies on the
-                // unset "ref". To double down on this, we do modify the initializer request. We're just using createGitInitializer
-                // for the authentication handling.
-                const { git, disposable } = await this.createGitInitializer({span}, workspace, {
-                    ...imgsrc.dockerFileSource,
-                    title: "irrelevant",
-                    ref: undefined,
-                }, user);
-                git.setCloneTaget(imgsrc.dockerFileSource.revision);
-                git.setTargetMode(CloneTargetMode.REMOTE_COMMIT);
-                git.setCheckoutLocation(".")
-                const source = new WorkspaceInitializer();
-                source.setGit(git);
+                let source: WorkspaceInitializer;
+                const disp = new DisposableCollection();
+                let checkoutLocation = this.getCheckoutLocation(workspace);
+                if (!imgsrc.dockerFileSource) {
+                    const {initializer, disposable} = await this.createInitializer({span}, workspace, workspace.context, user);
+                    source = initializer;
+                    disp.push(disposable);
+                } else {
+                    //TODO we cannot change this initializer structure now because it is part of how baserefs are computed in image-builder.
```

Comment:
nit: 
```suggestion
                    // TODO(se): we cannot change this initializer structure now because it is part of how baserefs are computed in image-builder.
```

see https://www.notion.so/gitpod/How-we-develop-e8ce7e562478458a9223eb9b797415b2#86e669f8e7df4f1289c9bede0855b5f9

Reply:
I'd propose we update the behaviour in image builder to accompany the new initializer. Also, the current change on how the baseRef is computed ignores the initializer entirely.